crates/messaging/src/implementors/natsio.rs

```diff
@@ -0,0 +1,74 @@
+use async_trait::async_trait;
+use nats::{Connection, Subscription};
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use tokio::runtime::Handle;
+use tokio::task::block_in_place;
+
+use anyhow::{Context, Result};
+use slight_common::BasicState;
+use slight_runtime_configs::get_from_state;
+
+use super::{PubImplementor, SubImplementor};
+
+#[derive(Clone, Debug)]
+pub struct NatsIoImplementor {
+    connection: Connection,
+    subscription_tokens: Arc<Mutex<HashMap<String, Subscription>>>,
+}
+
+impl NatsIoImplementor {
+    pub async fn new(slight_state: &BasicState) -> Self {
+        let nats_creds = get_from_state("NATS_CREDS", slight_state).await.unwrap();
+
+        let connection = nats::Options::with_static_credentials(&nats_creds)
+            .unwrap()
+            .connect("connect.ngs.global")
```

Comment:
What is "connect.ngs.global"?

Reply:
It's a public NATS server "connect.ngs.global" that is globally distributed – It's one of the examples for connectivity for the [nat's crate](https://crates.io/crates/nats) and what Bailey and I are using for our WasmIO demo (through Synadia – managed NATS).